app/init.go

```diff
@@ -0,0 +1,133 @@
+package app
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/app"
+	"github.com/iov-one/weave/coin"
+	"github.com/iov-one/weave/commands/server"
+	"github.com/iov-one/weave/crypto"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/migration"
+	"github.com/iov-one/weave/x/cash"
+	"github.com/iov-one/weave/x/currency"
+	"github.com/iov-one/weave/x/msgfee"
+	"github.com/iov-one/weave/x/multisig"
+	"github.com/iov-one/weave/x/validators"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/libs/log"
+)
+
+// GenInitOptions will produce some basic options for one rich
+// account, to use for dev mode
+func GenInitOptions(args []string) (json.RawMessage, error) {
+	// Your coins ticker code
+	code := "PRJCT"
+	if len(args) > 0 {
+		code = args[0]
+	}
+
+	var addr string
+	if len(args) > 1 {
+		addr = args[1]
+	} else {
+		// if no address provided, auto-generate one
+		// and print out a recovery phrase
+		bz, phrase, err := GenerateCoinKey()
+		if err != nil {
+			return nil, err
+		}
+		addr = bz.String()
+		fmt.Println(phrase)
+	}
+
+	type (
+		dict  map[string]interface{}
+		array []interface{}
+	)
+	collectorAddr, err := hex.DecodeString("3b11c732b8fc1f09beb34031302fe2ab347c5c14")
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot hex decode collector address")
+	}
+	return json.Marshal(dict{
+		"cash": array{
+			dict{
+				"address": addr,
+				"coins": array{
+					dict{
+						"whole":  123456789,
+						"ticker": code,
+					},
+				},
+			},
+		},
+		"currencies": array{
+			dict{
+				"ticker": "PRJCT",
+				"name":   "Your projects native token",
+			},
+		},
+
+		"conf": dict{
+			"cash": cash.Configuration{
+				CollectorAddress: collectorAddr,
+				MinimalFee:       coin.Coin{Whole: 0}, // no fee
+			},
+			"migration": dict{
+				"admin": addr,
+			},
+		},
+		"initialize_schema": []dict{
+			{"pkg": "cash", "ver": 1},
+			{"pkg": "sigs", "ver": 1},
+			{"pkg": "validators", "ver": 1},
+			{"pkg": "utils", "ver": 1},
+		},
+	})
+}
+
+// GenerateApp is used to create a stub for server/start.go command
+func GenerateApp(options *server.Options) (abci.Application, error) {
+	// db goes in a subdir, but "" -> "" for memdb
+	var dbPath string
+	if options.Home != "" {
+		dbPath = filepath.Join(options.Home, "abci.db")
+	}
+
+	stack := Stack(options.MinFee)
+	application, err := Application("project", stack, TxDecoder, dbPath, options.Debug)
+	if err != nil {
+		return nil, err
+	}
+
+	return DecorateApp(application, options.Logger), nil
+}
+
+// DecorateApp adds initializers and Logger to an Application
+func DecorateApp(application app.BaseApp, logger log.Logger) app.BaseApp {
+	application.WithInit(app.ChainInitializers(
+		&migration.Initializer{},
```

Comment:
Please update this to be in-line with the router/query/etc.

There are a few places that need to be kept in sync - good to add this to the docs.

When adding a package:
- `Stack` if it has a Decorator
- `Router` if it has a Handler
- `QueryRouter` if it has any Buckets
- `DecorateApp`/init if it has any Initializer.

Best to write down a list of all modules to use first, then make them all the same. As discussed above, I think the minimal set is:

* cash
* sigs
* multisig
* currency
* migration (as a runtime dependency of above modules)

`validators` is a nice to have to run a PoA testnet and good to have as a demo.

--

`msgfee` is a bit more complex and requires more complex Fee handling. I would not use this for default app.